### ISSUE TITLE
[BUG] Promote failed because fail to reach rancher-webhook

### ISSUE THREAD

**Issue by yaocw2020** [harvester/harvester#2187]
**Describe the bug**

I have a harvester environment with 4 hosts. After deleting the first management nodes, the promotion failed.
```
harvester-promote-harvester-v6pht-dw4xk                 0/1     Error       0          21h
harvester-promote-harvester-v6pht-ntfct                 0/1     Error       0          21h
harvester-promote-harvester-v6pht-x88qd                 0/1     Error       0          21h
```
```
harvester-jjmxp:/home/rancher # kubectl logs -f -n harvester-system harvester-promote-harvester-v6pht-dw4xk
Error from server (InternalError): Internal error occurred: failed calling webhook "default.machine.cluster.x-k8s.io": Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1beta1-machine?timeout=10s": context deadline exceeded
```

**To Reproduce**
1. Set up a 4 hosts harvester environment.
2. Delete the first management nodes.

**Expected behavior**

**Support bundle**

**Environment:**
 - Harvester ISO version: v1.0.1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal 

**Additional context**

**Comment by bk201** [comment]
This could be caused by non-ready rancher-webhook, when checking the environment the rancher-webhook is not ready.

**Comment by bk201** [comment]
We can verify this issue by applying the [workaround](https://github.com/harvester/harvester/issues/2191#issuecomment-1115794201).

**Comment by TachunLin** [comment]
Verified **fixed** on `master-7fcf04f8-head`(05/06). Close this issue.

### Result
When delete one management node, then promote the remaining work node failed with `SchedulingDisabled` 
We can apply this [workaround](https://github.com/harvester/harvester/issues/2191#issuecomment-1115794201) to promote the worker node correctly.

```
harvester-node-1:~ # kubectl get nodes
NAME               STATUS   ROLES                       AGE    VERSION
harvester-node-1   Ready    control-plane,etcd,master   158m   v1.21.11+rke2r1
harvester-node-2   Ready    control-plane,etcd,master   143m   v1.21.11+rke2r1
harvester-node-3   Ready    control-plane,etcd,master   126m   v1.21.11+rke2r1
```
![image](https://user-images.githubusercontent.com/29251855/167411004-ddf14a83-0621-4f1d-b9a6-9114d1fb032e.png)

### Test Information
* Test Environment: 4 nodes harvester on local kvm machine
* Harvester version: master-7fcf04f8-head (05/06)

### Verify Steps
1. Prepare a 4 nodes harvester cluster
2. Delete the first control plane node
  ```
  harvester-node-2:~ # kubectl get nodes 
  NAME               STATUS   ROLES                       AGE   VERSION
  harvester-node-0   Ready    control-plane,etcd,master   84m   v1.21.11+rke2r1
  harvester-node-1   Ready    control-plane,etcd,master   66m   v1.21.11+rke2r1
  harvester-node-2   Ready    control-plane,etcd,master   51m   v1.21.11+rke2r1
  harvester-node-3   Ready    <none>                      34m   v1.21.11+rke2r1
  
  harvester-node-2:~ # kubectl delete node harvester-node-0
  node "harvester-node-0" deleted
  ```
3. If the promotion fails, the promoting node should be stuck with SchedulingDisabled:
  ```
  harvester-node-2:~ # kubectl get nodes
  NAME               STATUS                     ROLES                       AGE    VERSION
  harvester-node-1   Ready                      control-plane,etcd,master   130m   v1.21.11+rke2r1
  harvester-node-2   Ready                      control-plane,etcd,master   115m   v1.21.11+rke2r1
  harvester-node-3   Ready,SchedulingDisabled   <none>                      98m    v1.21.11+rke2r1
  ```

4. Check the cluster condition 
  ```
  $kubectl describe clusters.cluster.x-k8s.io local -n fleet-local
  
  Status:
    Conditions:
      Last Transition Time:  2022-05-09T10:34:27Z
      Message:               Operation cannot be fulfilled on secrets "custom-598d8bf5678d-machine-plan": the object has been modified; please apply your changes to the latest version and try again
      Reason:                Error
      Status:                False
      Type:                  Ready
      Last Transition Time:  2022-05-09T09:10:02Z
      Status:                True
      Type:                  ControlPlaneInitialized
      Last Transition Time:  2022-05-09T10:34:27Z
      Message:               Operation cannot be fulfilled on secrets "custom-598d8bf5678d-machine-plan": the object has been modified; please apply your changes to the latest version and try again
      Reason:                Error
      Status:                False
      Type:                  ControlPlaneReady
      Last Transition Time:  2022-05-09T10:34:27Z
      Message:               Operation cannot be fulfilled on secrets "custom-598d8bf5678d-machine-plan": the object has been modified; please apply your changes to the latest version and try again
      Reason:                Error
      Status:                False
      Type:                  InfrastructureReady
    Control Plane Ready:     true
    Infrastructure Ready:    false
    Observed Generation:     91
    Phase:                   Provisioning
  Events:                    <none>
  ```

5. Find the machine ID from remaining control plan node
  ```
  harvester-node-2:~ # kubectl get machines.cluster.x-k8s.io -A
  NAMESPACE     NAME                  CLUSTER   NODENAME           PROVIDERID                PHASE     AGE    VERSION
  fleet-local   custom-598d8bf5678d   local     harvester-node-1   rke2://harvester-node-1   Running   135m   
  fleet-local   custom-853e845f95eb   local     harvester-node-2   rke2://harvester-node-2   Running   121m   
  fleet-local   custom-ee07120cb846   local     harvester-node-3   rke2://harvester-node-3   Running   103m
  ```

6. Use `node 2`
  ```
  harvester-node-2:~ # kubectl get secret custom-853e845f95eb-machine-plan -n fleet-local -o yaml | yq e '.metadata.labels."rke.cattle.io/machine-id"'
  e33d066bb6d80b0926d15754d3d5b5cb3baa57aee099caf4285d2daa78dbdd0
  ```

7. Set the machine ID to CAPI cluster object:
```
harvester-node-2:~ # kubectl label clusters.provisioning.cattle.io local -n fleet-local rke.cattle.io/init-node-machine-id=e33d066bb6d80b0926d15754d3d5b5cb3baa57aee099caf4285d2daa78dbdd0 --overwrite
cluster.provisioning.cattle.io/local labeled
```

8. Check The cluster should eventually reconcile and the promoting node should become a control plan node.
```
harvester-node-1:~ # kubectl get nodes
NAME               STATUS   ROLES                       AGE    VERSION
harvester-node-1   Ready    control-plane,etcd,master   158m   v1.21.11+rke2r1
harvester-node-2   Ready    control-plane,etcd,master   143m   v1.21.11+rke2r1
harvester-node-3   Ready    control-plane,etcd,master   126m   v1.21.11+rke2r1
```
